### PR TITLE
added kwargs for psf.convolve

### PR DESCRIPTION
**Pull request by maxecharles**
Allows for `"fft"` convolution method which is faster for medium-to-large arrays (thanks @ataras2)

### CODE REVIEW

**Comment by LouisDesdoigts** [comment]
This looks good but a few small tweaks are needed. 

1. I would change from a straight `**kwargs` to `method='auto'`. This is just to make it a bit more robust, since the only other option is precision, which we dont want to touch.
2. Docstring needs to be updated on the method
3. Unit test. Just under [this line](https://github.com/LouisDesdoigts/dLux/blob/main/tests/test_psfs.py#L23) add another convolve with the method input.

**Comment by maxecharles** [comment]
This good for the test?

```python
assert isinstance(psf.convolve(np.ones((2, 2)), method="fft"), PSF)
```

**Comment by LouisDesdoigts** [comment]
Yep more than happy for that as the test

File: src/dLux/psfs.py [hunk 1]
```diff
@@ -79,21 +79,26 @@ def downsample(self: PSF, n: int) -> PSF:
         """
         return self.set("data", dlu.downsample(self.data, n, mean=False))
 
-    def convolve(self: PSF, other: Array) -> PSF:
+    def convolve(self: PSF, other: Array, method="auto") -> PSF:
```
Comment (LouisDesdoigts) [inline src/dLux/psfs.py:82]: Last tweak - type hint for the method input: `method: str="auto")`

**Comment by LouisDesdoigts** [comment]
Last thing is the type hint for the method argument

**Comment by maxecharles** [comment]
I'm trying my best I promise

**Review by LouisDesdoigts — APPROVED** [review]
Looks good

**Pull request merged**